### ISSUE TITLE
ksize in max_pool

### ISSUE THREAD

**Issue by ayushchd** [tensorflow/tensorflow#4746]
I am trying to build a convolution (followed by max_pool) for variable length input size. Since the length of the input in a batch should be the same, I set the batch size to 1. However, for some reason ksize in tf.nn.max_pool is a list attribute required at run-time. This would prevent any neural network that uses variable input size to apply max pool. Is there a workaround for this? Should ksize be an input tensor otherwise?

**Comment by tatatodd** [comment]
FYI this was also asked about in #1957

We welcome contributions!

**Comment by ayushchd** [comment]
#1957 seems completely unrelated. Am I missing something?

**Comment by tatatodd** [comment]
I'm sorry, I meant #1967.  Indeed 1957 is completely unrelated.

**Comment by guotong1988** [comment]
I want to work on this issue

**Comment by guotong1988** [comment]
found something useful for the development.
http://stackoverflow.com/questions/33919948/how-to-set-adaptive-learning-rate-for-gradientdescentoptimizer

**Comment by girving** [comment]
It may be too late for @guoguo12's assistant, but if anyone else wants to jump on this I'm happy to answer questions.  The solution is to make a `MaxPoolV2` C++ op similar to `TopKV2`, and do the same for all the other convolutional ops.  We'd then deprecate the old op with `.Deprecated` and make the Python routines use the new version.  Here's `TopK` and `TopKV2` for comparison:

https://github.com/tensorflow/tensorflow/blob/master/tensorflow/core/ops/nn_ops.cc#L2013
https://github.com/tensorflow/tensorflow/blob/master/tensorflow/core/ops/nn_ops.cc#L2048

**Comment by guoguo12** [comment]
(You mean @guotong1988.)

**Comment by girving** [comment]
Whoops, sorry about that.

**Comment by yongtang** [comment]
Created a PR #9514 to try to address this issue. Would appreciate any review or comments.

**Comment by colaskirschoff** [comment]
@ayushchd I guess it's a bit late but there is a work around if you need to do your max_pool on a unique dimension. In this case you do not need any kernel size nor strides so you can use reduce_max directly.

**Comment by nkszjx** [comment]
pool = tf.reduce_max(activation, axis=1, keep_dims=True)
